### PR TITLE
Twitter API guide code

### PR DESCRIPTION
**Pull request by dastels**

### CODE REVIEW

File: Twitter_API/binascii.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+
+PAD = '='
+
+table_a2b_base64 = [
```
Comment (ladyada) [inline Twitter_API/binascii.py]: please convert to a bytestring :)

File: Twitter_API/binascii.py [hunk 2]
```diff
@@ -4,36 +4,30 @@
...
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-    -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1, -1,-1,-1,-1,
-]
-#pylint:enable=bad-whitespace
+#pylint:disable=bad-whitespace,invalid-name,bad-continuation
+table_a2b_base64 = bytes([
```
Comment (ladyada) [inline Twitter_API/binascii.py]: use the *actual* bytestring, type this into cpython to get the bytestring and paste it in

File: Twitter_API/code.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+the secrets dictionary must contain 'ssid' and 'password' at a minimum""")
+    raise
+
+def halt_and_catch_fire(message, *args):
+    """Log a critical error and stall the system."""
+    logger.critical(message, *args)
+    while True:
+        pass
+
+def connect_esp():
```
Comment (ladyada) [inline Twitter_API/code.py]: ? why do we need this again when its in pyportal library?

File: Twitter_API/code.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import busio
+from adafruit_pyportal import PyPortal
+from digitalio import DigitalInOut
+from adafruit_esp32spi import adafruit_esp32spi
+import adafruit_esp32spi.adafruit_esp32spi_requests as requests
+import adafruit_logging as logging
+
+username = 'codewisdom'
+
+logger = logging.getLogger('main')
```
Comment (ladyada) [inline Twitter_API/code.py]: i know you like logger stuff but lets keep it like the other pyportal demos and stick to print() :)

File: Twitter_API/code.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+bearer_token = get_bearer_token()
+logger.debug('Got it')
+
+headers = {'Authorization': 'Bearer ' + bearer_token}
+
+url = 'https://api.twitter.com/1.1/statuses/user_timeline.json?count=1&screen_name=' + username
+while True:
+    logger.debug('fetching latest tweet from @%s', username)
+    response = requests.get(url,
+                            headers=headers)
```
Comment (ladyada) [inline Twitter_API/code.py]: if you need to pass in headers, i think there's a pyportal PR that adds support? lets try that?

Comment (dastels) [reply Twitter_API/code.py]: I tried using just the PyPortal class, but had problems fetching the bearer token before the PyPortal ... ... I think I might have a way around that...

Comment (ladyada) [reply Twitter_API/code.py]: ok im interested if there's a way around it. if not, lets 'skip' the pyportal object and just DIY the whole thing with wifimanager & displayio

Comment (dastels) [reply Twitter_API/code.py]: OK... I saw no PRs for a set_headers so I just use `pyportal._headers = '...'`

Comment (ladyada) [reply Twitter_API/code.py]: oki!

Comment (dastels) [reply Twitter_API/code.py]: Not exactly pristine software design.. but hey... it works.  Maybe an optional headers param to fetch at some point?

Comment (ladyada) [reply Twitter_API/code.py]: if you want to add that to pyportal now i would merge it and then you could use that, seems handy!

Comment (dastels) [reply Twitter_API/code.py]: PR submitted to PyPortal adding get_headers

**Pull request merged**